x/delayedack/keeper/fraud.go

```diff
@@ -28,22 +29,43 @@ func (k Keeper) HandleFraud(ctx sdk.Context, rollappID string, ibc porttypes.IBC
 			"sequence", rollappPacket.Packet.Sequence,
 		}
 
+		pendingAddr := ""
+		transfer := rollappPacket.MustGetTransferPacketData()
 		if rollappPacket.Type == commontypes.RollappPacket_ON_ACK || rollappPacket.Type == commontypes.RollappPacket_ON_TIMEOUT {
-			// refund all pending outgoing packets
-			// we don't have access directly to `refundPacketToken` function, so we'll use the `OnTimeoutPacket` function
-			err := ibc.OnTimeoutPacket(ctx, *rollappPacket.Packet, rollappPacket.Relayer)
-			if err != nil {
-				logger.Error("failed to refund reverted packet", append(logContext, "error", err.Error())...)
-			}
+			// for sent packets, we restore the packet commitment
+			// the packet will be handled over the new rollapp revision
+			commitment := channeltypes.CommitPacket(k.cdc, rollappPacket.Packet)
+			k.channelKeeper.SetPacketCommitment(ctx, rollappPacket.Packet.SourcePort, rollappPacket.Packet.SourceChannel, rollappPacket.Packet.Sequence, commitment)
+			pendingAddr = transfer.Sender
+		} else {
+			// for incoming packets, we need to reset the packet receipt
+			ibcPacket := rollappPacket.Packet
+			k.deletePacketReceipt(ctx, ibcPacket.GetDestPort(), ibcPacket.GetDestChannel(), ibcPacket.GetSequence())
+			pendingAddr = transfer.Receiver
+		}
+
+		// delete the packet
+		err := k.DeleteRollappPacket(ctx, &rollappPacket)
+		if err != nil {
+			logger.Error("failed to delete reverted packet", append(logContext, "error", err.Error())...)
+			continue
 		}
-		// Update status to reverted
-		_, err := k.UpdateRollappPacketWithStatus(ctx, rollappPacket, commontypes.Status_REVERTED)
+
+		// delete the pending packet
+		err = k.DeletePendingPacketByAddress(ctx, pendingAddr, rollappPacket.RollappPacketKey())
 		if err != nil {
-			logger.Error("error reverting IBC rollapp packet", append(logContext, "error", err.Error())...)
-			return err
+			logger.Error("failed to delete reverted PacketByAddress", append(logContext, "error", err.Error())...)
+			continue
 		}
```

Comment:
as said in other places, better to return error no?

Reply:
it actually a "must" call as it's only modifies store. no recoverable error should be returned
changed to must